### PR TITLE
Enabe IAM API in promoter projects and test signing svc acct

### PR DESCRIPTION
**Pull request by puerco**
This PR adds two things to the infra bash scripts:

1. It adds to the `ensure-prod-storage.sh` script calls to enable the
IAM Service Account Credentials API (iamcredentials.googleapis.com)
in the promoter projects (k8s-cip-test-prod and k8s-artifacts-prod). This is required to generate OIDC identity tokens that will be used to sign images and artifacts.

2. It adds a new service account to the test project (`k8s-infra-promoter-test-signer`) to sign images during e2es and ensures the account that runs the tests has access to them.

Fixes https://github.com/kubernetes/k8s.io/issues/3445

/sig k8s-infra
/sig release

/assign @ameukam 
/cc @kubernetes/release-engineering 

Signed-off-by: Adolfo García Veytia <adolfo.garcia@uservers.net>

### CODE REVIEW

**Comment by puerco** [comment]
If it looks ok, could you actuate after merging @ameukam ? :pray:

File: infra/gcp/bash/ensure-prod-storage.sh [hunk 1]
```diff
@@ -391,6 +391,28 @@ function ensure_all_prod_special_cases() {
...
         "group:k8s-infra-gcs-access-logs@kubernetes.io" \
         "objectViewer"
+
+    # Special case: In order to run the container image signing tests, k8s-cip-test-prod
+    # requires to have the IAM Service Account Credentials API enabled to generate OIDC
+    # identity tokens. The production project needs the API too to sign all promoted images.
+
+    color 6 "Ensuring IAM Service Account Credentials API for image signing"
+    ensure_services "${IMAGE_PROMOTER_TEST_PROD_PROJECT}" "iamcredentials.googleapis.com"
+    ensure_services "${PROD_PROJECT}" "iamcredentials.googleapis.com"
```
Comment (ameukam) [inline infra/gcp/bash/ensure-prod-storage.sh:401]: Why enable this service for `k8s-artifacts-prod` ? do we need to get tokens after promotion is done ?

Comment (puerco) [reply infra/gcp/bash/ensure-prod-storage.sh:401]: Correct! Once images are copied to the target registries, the promoter will need to generate tokens with the identity of the Main Kubernetes Signer. I've opened an issue to better visualize the flows. This PR is part of the Promotion and e2e flows:

https://github.com/kubernetes/release/issues/2445

**Comment by puerco** [comment]
(repushed as I found a bug in the token creator principal)

**Pull request merged**